src/App.tsx

```diff
@@ -1,61 +1,116 @@
-import React from 'react';
-import 'bulma/css/bulma.css';
-import './App.scss';
+import React, { useEffect, useState } from "react";
+import "bulma/css/bulma.css";
+import "./App.scss";
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  "Dumplings",
+  "Carrot",
+  "Eggs",
+  "Ice cream",
+  "Apple",
+  "Bread",
+  "Fish",
+  "Honey",
+  "Jam",
+  "Garlic",
 ];
 
+export function getReorderedGoods(
+  goods: string[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1: string, good2: string): number => {
+    if (sortType === "ALPHABET") {
+      return good1.localeCompare(good2);
+    }
+
+    if (sortType === "LENGTH") {
+      return good1.length - good2.length;
+    }
+
+    if (sortType === "NONE") {
+      return 0;
+    }
+
+    return 0;
+  });
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState("NONE");
+  const [isReversed, setIsReversed] = useState(false);
+  const goodies = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+  useEffect(() => {}, [isReversed]);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            sortType !== "ALPHABET"
+              ? "button is-info is-light"
+              : "button is-info"
+          }
```

Comment:
use ClassNames library for better styles conditions

Reply:
how about now?